tests/infra/device_connector.py

```diff
@@ -0,0 +1,110 @@
+# SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+from enum import Enum
+from typing import Sequence
+
+import jax
+import jax._src.xla_bridge as xb
+
+
+class DeviceType(Enum):
+    CPU = "cpu"
+    TT = "tt"
+    GPU = "gpu"
+
+
+class DeviceConnector:
+    """
+    Singleton class providing connections to devices on which jax commands will be
+    executed.
+
+    As a singleton it is instantiated only once, thus making sure that PJRT plugin is
+    registered exactly once. Registration needs to happen before any other jax commands
+    are executed. Registering it multiple times would cause error.
+
+    Do not instantiate this class. Use provided global instance.
+    """
+
+    _instance = None
+
+    def __new__(cls, *args, **kwargs):
+        # Ensure that only one instance of the class is created.
+        if cls._instance is None:
+            cls._instance = super().__new__(cls, *args, **kwargs)
+
+        return cls._instance
+
+    def __init__(
+        self, tt_pjrt_plugin_relpath: str = "build/src/tt/pjrt_plugin_tt.so"
```

Comment:
Should the string be put in some global variable instead of just here plainly? Something like:
TT_PJRT_PLUGIN_PATH="build/src/tt/pjrt_plugin_tt.so"